crates/cargo-wdk/src/actions/package/mod.rs

```diff
@@ -0,0 +1,384 @@
+            {
+                debug!(
+                    "Found atleast one valid Rust project directory: {}, continuing with the \
+                     package flow",
+                    dir.path()
+                        .file_name()
+                        .expect("error reading the folder name")
+                        .to_string_lossy()
+                );
+                is_valid_dir_with_rust_projects = true;
+                break;
+            }
+        }
+
+        if !is_valid_dir_with_rust_projects {
+            return Err(PackageProjectError::NoValidRustProjectsInTheDirectory(
+                self.working_dir.clone(),
+            ));
+        }
+
+        debug!("Iterating over each dir entry and process valid Rust(possibly driver) projects");
+        let mut did_fail_atleast_one_project = false;
+        for dir in dirs {
+            debug!(
+                "Verifying the dir entry if it is a valid Rust project: {}",
+                dir.path().display()
+            );
+            if dir.file_type()?.is_dir() && self.fs_provider.exists(&dir.path().join("Cargo.toml"))
+            {
+                info!(
+                    "Processing Rust(possibly driver) project: {}",
+                    dir.path()
+                        .file_name()
+                        .expect("error reading the folder name")
+                        .to_string_lossy()
+                );
+                match self.get_cargo_metadata(&dir.path()) {
+                    Ok(cargo_metadata) => {
+                        if let Err(e) = self.run_from_workspace_root(&dir.path(), &cargo_metadata) {
+                            did_fail_atleast_one_project = true;
+                            log_error!(
+                                "Error packaging the child project: {}, error: {}",
+                                dir.path()
+                                    .file_name()
+                                    .expect("error reading the folder name")
+                                    .to_string_lossy(),
+                                e
+                            );
+                        }
+                    }
+                    Err(e) => {
+                        did_fail_atleast_one_project = true;
+                        log_error!("Error reading cargo metadata: {}", e);
+                    }
+                }
+            } else {
+                debug!("Skipping the dir entry as it is not a valid Rust project");
+            }
+        }
+
+        debug!("Done checking for valid Rust(possibly driver) projects in the working director");
+        if did_fail_atleast_one_project {
+            return Err(PackageProjectError::OneOrMoreRustProjectsFailedToBuild(
+                self.working_dir.clone(),
+            ));
+        }
+
+        info!("Building and packaging completed successfully");
+        Ok(())
+    }
+
+    fn run_from_workspace_root(
+        &self,
+        working_dir: &PathBuf,
```

Comment:
Change this `&PathBuf` to `&Path`

Reply:
I believe this will be touched as part of https://github.com/svasista-ms/windows-drivers-rs/pull/6. Let's revisit this after that PR merge.

Reply:
Looks like this wasn't resolved in Shravan's PR.